fn-encapsulate.js

```diff
@@ -0,0 +1,42 @@
+var path = require('path');
+var crypto = require('crypto');
+var fs = require('fs');
+
+var Module = module.constructor;
+
+module.exports = function(functionFilename, expectedFunctionName) {
```

Comment:
Interesting implementation. I was not aware that you can create modules like this. Is this official API, or is this something that's internal and might change without being announced?

Also why not just do a eval(fs.readFileSync('filename'))?